### ISSUE TITLE
event flow sometimes stops on high flow rate and can only be restored by flushing all the keys in the redis DB

### ISSUE THREAD

**Issue by drook** [getsentry/sentry#23145]
## Important Details

Instance: On-Premise wo/ Docker (standalone), version 20.8.0

## Description
Sometimes the event flow just stops, and no restart of relay/snuba/sentry/sentry celery workers allthogether help. The only thing that actually helps is FLUSHALL in redis sentry database. Then the event flow recovers just instantly. This usually happens when the event flow rate exceeds certain amount of events, say like 20K per minute (last observed case).

I fully realize that this is a lousy ticket, one of those that rarely gets attention from developers, because it doesn't really gives any diagnostics except stating the fact. But the thing is - logs are silent about this, so I just don't know what diag should I provide.

**Comment by BYK** [comment]
This means your instance cannot catch up with the event rate. This thread may help you: https://forum.sentry.io/t/how-to-clear-backlog-and-monitor-it/10715/15?u=byk

Also I recommend upgrading to something newer than `20.10` as they have a fix that reduces the pressure on Redis when under load. The latest version we have is `21.1` which was released last Friday.

**Comment by BYK** [comment]
@drook do you think we answered your question?

**Comment by drook** [comment]
Yeah, thanks, totally. I started to test-migrate to around 21.x, there was an issue with new rust compiler I got, then guys told me that they've fixed these, but atm I was overwhelmed by new priority work (saying this to state that I'm now just a dweller from the depths that reports random bugs and disappears, - I'm using sentry and relying on it), so I'm still excited to try this new version and will do that asap. I think it's safe to close this for now, thanks.
